pkg/cloud/common/resources.go

```diff
@@ -0,0 +1,55 @@
+package common
+
+import (
+	"fmt"
+	"strings"
+
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/config"
+)
+
+const (
+	CloudControllerManagerProviderLabel = "infrastructure.openshift.io/cloud-controller-manager"
+	CloudNodeManagerCloudProviderLabel  = "infrastructure.openshift.io/cloud-node-manager"
+)
+
+func GetCommonResources(config config.OperatorConfig) ([]client.Object, error) {
+	commonResources := make([]client.Object, 0, 1)
+	if !config.IsSingleReplica {
+		pdb, err := getPDB(config)
+		if err != nil {
+			return nil, err
+		}
+		commonResources = append(commonResources, pdb)
+	}
+	return commonResources, nil
+}
+
+func getPDB(config config.OperatorConfig) (*policyv1.PodDisruptionBudget, error) {
+	minAvailable := intstr.FromInt(1)
```

Comment:
Change this to `maxUnavailable` maybe? Then it would work even with a single replica?

Reply:
If we will strictly constraint number of ccm replicas to two, yes it might work with single replica.

However, i don't like this way, feel that it's harder to understand and troubleshoot then.

Reply:
`maxUnavailable` tends to be more widely recommended as it is more flexible to changes than `minAvailable` and is in general safer.

In my experience, `minAvailable` is more likely to cause an issue than `maxUnavailable`. IMO this should be `maxUnavailable`, even if we have 3 replicas, yes, it slows down rollouts, but that's not necessarily a bad thing for such an important component